x-pack/plugins/enterprise_search/server/lib/engines/fetch_indices_stats.ts

```diff
@@ -9,21 +9,26 @@ import { IScopedClusterClient } from '@kbn/core-elasticsearch-server/src/client/
 
 import { EnterpriseSearchEngineIndex } from '../../../common/types/engines';
 
-export const fetchIndicesStats = async (client: IScopedClusterClient, indices: string[]) => {
-  const { indices: indicesStats = {} } = await client.asCurrentUser.indices.stats({
-    index: indices,
-    metric: ['docs'],
-  });
-
-  const indicesWithStats = indices.map((indexName: string) => {
-    const indexStats = indicesStats[indexName];
-    const hydratedIndex: EnterpriseSearchEngineIndex = {
-      count: indexStats?.primaries?.docs?.count ?? 0,
-      health: indexStats?.health ?? 'unknown',
-      name: indexName,
-    };
-    return hydratedIndex;
-  });
-
-  return indicesWithStats;
+const fetchIndexStats = async (client: IScopedClusterClient, index: string) => {
+  try {
+    const resp = await client.asCurrentUser.indices.stats({ index, metric: ['docs'] });
+    return resp.indices?.[index] ?? {};
+  } catch {
+    return {};
+  }
 };
+
+export const fetchIndicesStats = async (
+  client: IScopedClusterClient,
+  indices: string[]
+): Promise<EnterpriseSearchEngineIndex[]> =>
+  Promise.all(
+    indices.map(async (index: string) => {
+      const stats = await fetchIndexStats(client, index);
+      return {
+        count: stats.primaries?.docs?.count ?? 0,
```

Comment:
I don't remember if we wanted to display 0 or `unknown` for doc count. CC: @julianrosado

Reply:
fwiw: this keeps the [existing behavior](https://github.com/elastic/kibana/pull/156571/files#diff-1ee8a8ab975b3bd51c0c25f64b5e5faf51651cd62cccddbfbe0f80e42530c0dcL21) for counts and missing indices

Reply:
pretty easy to just make this property nullable though. produces this in the UI:

<img width="695" alt="image" src="https://user-images.githubusercontent.com/1699281/235978319-b49c04ae-e5c4-4f98-bf4a-569649136ef7.png">

Reply:
Not sure if it was previously 0 for a specific reason though. Sorting? Even in that case it doesn't make sense (to me).

Reply:
> Not sure if it was previously 0 for a specific reason though. Sorting? Even in that case it doesn't make sense (to me).

probably just a typing convenience oh the part of the engineer. it was easy enough to change ✅

Reply:
I remember checking this when I added this change, I tried to replicate the same response received from ent-search when doc count is 0. May be missed from my part.